extensions/BugModal/lib/WebService.pm

```diff
@@ -62,6 +81,26 @@ sub rest_resources {
     ]
 }
 
+sub products {
+    my $user = Bugzilla->user;
+    my @products = @{ $user->get_enterable_products };
+
+    @products = map { { name => $_->name } } @products;
+    return {
+        products     => \@products,
+    };
+}
+
+sub components {
+    my ($self, $params) = @_;
+    my $product = Bugzilla::Product->check({ name => $params->{product_name}, cache => 1 });
+    
+    my @components = map { { name => $_->name } } @{ $product->components };
```

Comment:
But I think I do know a useful question to offer. What is the equivalent function for ```_name()``` that returns Products, rather than Components, in the same structure as ```_name()``` does?